warehouse/src/main/java/io/skai/warehouse/cache/ComponentCacheStore.java

```diff
@@ -0,0 +1,57 @@
+package io.skai.warehouse.cache;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import io.skai.warehouse.model.component.Component;
+import io.skai.warehouse.repository.ComponentPersistence;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+
+public class ComponentCacheStore {
+
+    ComponentPersistence componentPersistence;
+
+    private final LoadingCache<Long, Component> cache;
+
+    public ComponentCacheStore(int expiryDuration, TimeUnit timeUnit, ComponentPersistence componentPersistence) {
+        this.componentPersistence = componentPersistence;
+        cache = CacheBuilder.newBuilder()
+                .expireAfterWrite(expiryDuration, timeUnit)
+                .build(getCacheLoader());
+    }
+
+    public Optional<Component> get(Long key) {
+        return Optional.ofNullable(cache.getUnchecked(key));
+    }
+
+    public void add(Long key, Component value) {
```

Comment:
you shouldn't put manually, please check https://github.com/kenshoo/apple-api/blob/master/main/src/main/java/io/skai/apple/api/client/impl/AppleAuthClientImpl.java